samples/sync-history/src/main/java/org/mozilla/samples/sync/history/MainActivity.kt

```diff
@@ -149,21 +135,48 @@ class MainActivity : AppCompatActivity(), LoginFragment.OnLoginCompleteListener,
         }
     }
 
-    private fun displayAndPersistProfile(code: String, state: String) {
-        launch {
-            account.completeOAuthFlow(code, state).await()
-            val profile = account.getProfile().await()
-            displayProfile(profile)
-            account.toJSONString().let {
-                getSharedPreferences(FXA_STATE_PREFS_KEY, Context.MODE_PRIVATE)
-                        .edit().putString(FXA_STATE_KEY, it).apply()
+    private val accountObserver = object : AccountObserver {
+        override fun beginOathFlow(authUrl: String) {
```

Comment:
We still plan on the `FirefoxAccountFeature`, right? It would observe these events e.g. `onBeginOAuthFlow` and open a new tab by default?

Reply:
Yes, as an optional feature. I won't upgrade sample apps to use it, but we can use it in r-b.